File: java/program/Perangkat.java
// Class Perangkat inheriting from Elektronik
public class Perangkat extends Elektronik {
    private String merek;
    private String kode_sku;
    int garansi; // in months

    // Constructor without parameters
    public Perangkat() {
        super();
        this.merek = "";
        this.kode_sku = "";
        this.garansi = 0;
    }

    // Constructor with parameters
    public Perangkat(String id, String nama, int harga, String merek, String kode_sku, int garansi) {
        super(id, nama, harga);
        this.merek = merek;
        this.kode_sku = kode_sku;
        this.garansi = garansi;
    }

    // Getter & Setter for merek
    public String getMerek() {return merek;}
    public void setMerek(String merek) {this.merek = merek;}

    // Getter & Setter for kode_sku
    public String getKodeSKU() {return kode_sku;}
    public void setKodeSKU(String kode_sku) {this.kode_sku = kode_sku;}

    // Getter & Setter for garansi
    public int getGaransi() {return garansi;}
    public void setGaransi(int garansi) {this.garansi = garansi;}
}

File: python/program/Perangkat.py
from Elektronik import Elektronik

### Inheritance from Elektronik class
class Perangkat(Elektronik):
    ## Constructor
    def __init__(self, id="", nama="", harga=0, merek="", kode_sku="", garansi=0):
        super().__init__(id, nama, harga)
        self.merek = merek
        self.kode_sku = kode_sku 
        self.garansi = garansi

    ## Setter and getter for merek
    def set_merek(self, merek): self.merek = merek
    def get_merek(self): return self.merek  

    ## Setter and getter for kode_sku
    def set_kode_sku(self, kode_sku): self.kode_sku = kode_sku
    def get_kode_sku(self): return self.kode_sku    

    ## Setter and getter for garansi
    def set_garansi(self, garansi): self.garansi = garansi
    def get_garansi(self): return self.garansi